### PR TITLE
Have flying actors account for terrain height

### LINKED ISSUE #8547: Height issue with TS aircraft [$69]
I'll let the screenshot talk:
height-error-ts

<bountysource-plugin>
---
There is a **[$69 open bounty](https://www.bountysource.com/issues/22772145-height-issue-with-ts-aircraft?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github)** on this issue. Add to the bounty at [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by abcdefg30**
Adresses #8547 partly.

### CODE REVIEW

File: OpenRA.Mods.Common/Activities/Air/Fly.cs [hunk 1]
```diff
@@ -37,6 +37,9 @@ public Fly(Actor self, Target t, WDist minRange, WDist maxRange)
 
 		public static void FlyToward(Actor self, Plane plane, int desiredFacing, WDist desiredAltitude)
 		{
+			var cell = self.World.Map.CellContaining(plane.CenterPosition);
+			desiredAltitude = desiredAltitude + new WDist(self.World.Map.MapHeight.Value.Contains(cell) ? 512 * self.World.Map.MapHeight.Value[cell] : 0);
```
Comment (pchote) [inline OpenRA.Mods.Common/Activities/Air/Fly.cs]: These need to use `map.DistanceAboveTerrain` instead of rolling their own checks.

Comment (abcdefg30) [inline OpenRA.Mods.Common/Activities/Air/Fly.cs]: Done.

**Comment by Mailaender** [comment]
Helicopters behave a bit annoying, but I count this as an improvement. No side effects on the non-height-mapped mods. :white_check_mark:

**Comment by Mailaender** [comment]
Does not fix #8925 though (actually zero effect on it).

**Comment by RoosterDragon** [comment]
:+1: The helicopters are indeed a little silly but I'm sure some smoothing can be worked out later.

**Comment by pchote** [comment]
@RoosterDragon: I have a branch for that (™).

**Comment by penev92** [comment]
Helis are indeed stupid, but planes work pretty nicely :+1:

**Comment by penev92** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-%28bleed%29/_compare/6edfd2f0f33c9c656b2517eb8db1ad88724b3047...c53b52c1fa5aaa3993a78f9fc2978bae638d8df6)

**Pull request merged**